Tests fail on aarch64: test/unit/helpers.lua:254: declaration specifier expected near '__uint128_t' at line 396

### Neovim version (nvim -v)

0.7.0

### Vim (not Nvim) behaves the same?

no, Lua test suite, not applicable for vim

### Operating system/version

Linux, openSUSE/Tumbleweed as of 2022-04-19

### Terminal name/version

N/A

### $TERM environment variable

N/A

### Installation

openSUSE packages

### How to reproduce the issue

Run the test suite (`make unittest`).

### Expected behavior

Tests pass.

### Actual behavior

```
[  163s] /usr/bin/make  -f CMakeFiles/unittest.dir/build.make CMakeFiles/unittest.dir/build
[  163s] make[3]: Entering directory '/home/abuild/rpmbuild/BUILD/neovim-0.7.0/build'
[  163s] /usr/bin/cmake -DBUSTED_PRG=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/lj-busted.sh -DLUA_PRG=/usr/bin/luajit -DWORKING_DIR=/home/abuild/rpmbuild/BUILD/neovim-0.7.0 -DBUSTED_OUTPUT_TYPE=nvim -DTEST_DIR=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/test -DBUILD_DIR=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/build -DTEST_TYPE=unit -P /home/abuild/rpmbuild/BUILD/neovim-0.7.0/cmake/RunTests.cmake
[  163s] -- Tests exited non-zero: 1
[  163s] -- Output to stderr:
[  163s] + /usr/bin/luajit /usr/bin/busted -v -o test.busted.outputHandlers.nvim --lazy --helper=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/test/unit/preload.lua '--lpath=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/build/?.lua' '--lpath=/home/abuild/rpmbuild/BUILD/neovim-0.7.0/runtime/lua/?.lua' '--lpath=?.lua' test/unit
[  163s] test/unit/helpers.lua:254: declaration specifier expected near '__uint128_t' at line 396
[  163s] /usr/bin/luajit: /usr/share/lua/5.1/busted/runner.lua:104: attempt to concatenate local 'message' (a table value)
[  163s] stack traceback:
[  163s]        /usr/share/lua/5.1/busted/runner.lua:104: in function 'fn'
[  163s]        /usr/share/lua/5.1/mediator.lua:103: in function 'publish'
[  163s]        /usr/share/lua/5.1/busted/modules/helper_loader.lua:23: in function 'helperLoader'
[  163s]        /usr/share/lua/5.1/busted/runner.lua:161: in function </usr/share/lua/5.1/busted/runner.lua:11>
[  163s]        /usr/bin/busted:3: in main chunk
[  163s]        [C]: at 0xaaaae92f2074
[  163s]
[  163s] CMake Error at /home/abuild/rpmbuild/BUILD/neovim-0.7.0/cmake/RunTests.cmake:98 (message):
[  163s]   unit tests failed with error: 1
[  163s]
[  163s]
[  163s] make[3]: *** [CMakeFiles/unittest.dir/build.make:73: CMakeFiles/unittest] Error 1
[  163s] make[3]: Leaving directory '/home/abuild/rpmbuild/BUILD/neovim-0.7.0/build'
[  163s] make[2]: *** [CMakeFiles/Makefile2:557: CMakeFiles/unittest.dir/all] Error 2
[  163s] make[1]: *** [CMakeFiles/Makefile2:564: CMakeFiles/unittest.dir/rule] Error 2
[  163s] make: *** [Makefile:232: unittest] Error 2
```

[Complete build log](https://github.com/neovim/neovim/files/8514628/_log.txt) with lists of all packages used and steps taken to reproduce.

I have been pointed to https://github.com/neovim/neovim/issues/7423 (and https://github.com/neovim/neovim/pull/7657) which certainly resembles this situation but for `__uint128_t` instead of `__float128`.

--- a/test/unit/helpers.lua
+++ b/test/unit/helpers.lua
@@ -148,7 +148,11 @@ local cdef = ffi.cdef
 
 local cimportstr
 
-local previous_defines_init = ''
+local previous_defines_init = [[
+typedef struct { char bytes[16]; } __attribute__((aligned(16))) __uint128_t;
+typedef struct { char bytes[16]; } __attribute__((aligned(16))) __float128;
+]]
+
 local preprocess_cache_init = {}
 local previous_defines_mod = ''
 local preprocess_cache_mod = nil

